java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -1155,6 +1164,10 @@ private static long copySlicedAndPad(DataWriter out,
   // VALIDITY
   /////////////////////////////////////////////
 
+  private static boolean needsValidityBuffer(long nullCount) {
+    return nullCount > 0 || nullCount == ColumnView.UNKNOWN_NULL_COUNT;
+  }
```

Comment:
I am a bit nervous with this. Can the column be non-nullable if its null count is `UNKNOWN_NULL_COUNT`?

Reply:
No, because we already check if the sliced column came from a column with a null count of zero. In that case we know the null count of the sliced column as well (also zero). So it should only be unknown when we know there’s at least one null in the original column that was sliced, and therefore the validity buffer is present.